src/nvim/highlight_group.c

```diff
@@ -135,263 +135,321 @@ static const char e_missing_argument_str[]
+  "DiffText             guifg=NvimLightGrey1  ctermfg=255   guibg=NvimDarkCyan   ctermbg=30",
+  "Directory            guifg=NvimLightCyan   ctermfg=123",
+  "ErrorMsg             guifg=NvimLightRed    ctermfg=217",
+  "FloatShadow          guibg=NvimDarkGrey1   ctermbg=232   blend=80",
+  "FloatShadowThrough   guibg=NvimDarkGrey1   ctermbg=232   blend=100",
+  "Folded               guifg=NvimLightGrey4  ctermfg=247   guibg=NvimDarkGrey3  ctermbg=236",
+  "LineNr               guifg=NvimDarkGrey4   ctermfg=239",
+  "MatchParen           guibg=NvimDarkGrey4   ctermbg=239   gui=bold             cterm=bold",
+  "ModeMsg              guifg=NvimLightGreen  ctermfg=158",
+  "MoreMsg              guifg=NvimLightCyan   ctermfg=123",
+  "NonText              guifg=NvimDarkGrey4   ctermfg=239",
+  "NormalFloat          guifg=NvimLightGrey2  ctermfg=253   guibg=NvimDarkGrey1  ctermbg=232",
+  "Pmenu                guifg=NvimLightGrey2  ctermfg=253   guibg=NvimDarkGrey3  ctermbg=236",
+  "PmenuSel             guifg=NvimDarkGrey3   ctermfg=236   guibg=NvimLightGrey2 ctermbg=253 blend=0",
+  "PmenuThumb           guibg=NvimDarkGrey4   ctermbg=239",
+  "Question             guifg=NvimLightCyan   ctermfg=123",
+  "RedrawDebugClear     guibg=NvimDarkCyan    ctermbg=30",
+  "RedrawDebugComposed  guibg=NvimDarkGreen   ctermbg=22",
+  "RedrawDebugRecompose guibg=NvimDarkRed     ctermbg=52",
+  "Search               guifg=NvimLightGrey1  ctermfg=255   guibg=NvimDarkYellow ctermbg=58",
+  "SignColumn           guifg=NvimDarkGrey4   ctermfg=239",
+  "SpecialKey           guifg=NvimDarkGrey4   ctermfg=239",
+  "SpellBad             guisp=NvimLightRed    gui=undercurl cterm=undercurl",
+  "SpellCap             guisp=NvimLightYellow gui=undercurl cterm=undercurl",
+  "SpellLocal           guisp=NvimLightGreen  gui=undercurl cterm=undercurl",
+  "SpellRare            guisp=NvimLightCyan   gui=undercurl cterm=undercurl",
+  "StatusLine           guifg=NvimLightGrey3  ctermfg=251   guibg=NvimDarkGrey1  ctermbg=232",
+  "StatusLineNC         guifg=NvimLightGrey4  ctermfg=247   guibg=NvimDarkGrey1  ctermbg=232",
+  "TabLine              guifg=NvimLightGrey3  ctermfg=251   guibg=NvimDarkGrey1  ctermbg=232",
+  "Visual               guibg=NvimDarkGrey4   ctermbg=239",
+  "WarningMsg           guifg=NvimLightYellow ctermfg=217",  // In 256 colors fall back to red
+
+  // Syntax
+  "Comment    guifg=NvimLightGrey4   ctermfg=247",
+  "Constant   guifg=NvimLightGrey2   ctermfg=253",
+  "String     guifg=NvimLightGreen   ctermfg=158",
+  "Identifier guifg=NvimLightBlue    ctermfg=NONE",  // No fallback in 256 colors to reduce noise
+  "Function   guifg=NvimLightCyan    ctermfg=123",
+  "Statement  guifg=NvimLightGrey2   ctermfg=253 gui=bold          cterm=bold",
+  "Operator   guifg=NvimLightGrey2   ctermfg=253",
+  "PreProc    guifg=NvimLightGrey2   ctermfg=253",
+  "Type       guifg=NvimLightGrey2   ctermfg=253",
+  "Special    guifg=NvimLightGrey2   ctermfg=253",
```

Comment:
`diffRemoved` links to `Special`, so removed lines in `diff` file type don't get highlighting. Not sure if that's worth handling in the default colorscheme, though.
![image](https://github.com/neovim/neovim/assets/19489738/72de5eea-6d06-4f22-8c78-33fbeab0a972)

Reply:
Yeah, the syntax files are kinda all over the place with the highlight groups -- this is one of the things we want to do better with treesitter, lessons learned. 

That being said, it may be worth contacting the syntax maintainers about links that can clearly be improved (like this one, which probably should link to DiffRemoved instead).